.github/workflows/vendor_release.yml

```diff
@@ -0,0 +1,31 @@
+name: Generate vendored content for the release
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  generate-and-publish-vendored-code:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Generate cargo vendored tarball
+        run: |
+          tag=$(echo $GITHUB_REF | sed 's|refs/tags/||')
+          tarball_name="guest-components-${tag}-vendor.tar.gz"
+          vendor_dir_list=""
+          for i in $(find . -name 'Cargo.lock'); do
```

Comment:
SInce this is a workspace, shouldn't there be exactly 1 Cargo.lock file in the project root?

Reply:
I'd assume it's unlikely that we would revert it back to multiple individual projects. However what we might want to cover is different feature toggles, or would the default features be sufficient?

Reply:
> However what we might want to cover is different feature toggles, or would the default features be sufficient?

That's a very good question, and I would start with what I proposed, as https://doc.rust-lang.org/cargo/commands/cargo-vendor.html does not mention anything about specific features.